File: sdk/java/src/main/java/com/yowyob/loyalty/sdk/ApiException.java
package com.yowyob.loyalty.sdk;

import java.util.Map;

/** Erreur renvoyée par l'API Loyalty (statut HTTP non 2xx). */
public class ApiException extends LoyaltyException {

    private final int statusCode;
    private final transient Map<String, Object> body;

    public ApiException(String message, int statusCode, Map<String, Object> body) {
        super(message);
        this.statusCode = statusCode;
        this.body = body;
    }

    public int getStatusCode() {
        return statusCode;
    }

    public Map<String, Object> getBody() {
        return body;
    }
}

File: sdk/php/src/Exception/ApiException.php
<?php

namespace Yowyob\Loyalty\Exception;

/**
 * Erreur renvoyée par l'API Loyalty (statut HTTP non 2xx).
 */
class ApiException extends LoyaltyException
{
    private int $statusCode;
    private ?array $body;

    public function __construct(string $message, int $statusCode, ?array $body = null)
    {
        parent::__construct($message, $statusCode);
        $this->statusCode = $statusCode;
        $this->body = $body;
    }

    public function getStatusCode(): int
    {
        return $this->statusCode;
    }

    public function getBody(): ?array
    {
        return $this->body;
    }
}
